app/components/file-viewer/index.jsx

```diff
@@ -20,39 +21,52 @@ const VIEWERS = {
   image: ImageViewer,
   text: TextViewer,
   video: VideoPlayer,
-  audio: AudioPlayer
+  audio: AudioPlayer,
+  canvas: CanvasViewer
 };
 
 function subjectViewerSelector(props) {
   if (Array.isArray(props.type)) {
     if (props.type.includes('audio')) {
       return VIEWERS.audio;
     }
-    // ... add outher here if neccessary
+    // ... add other here if neccessary
   }
   return VIEWERS[props.type] || DefaultViewer;
 }
 
 function FileViewer(props) {
   const Viewer = subjectViewerSelector(props);
-
+  const viewerProps = {
+    className: props.className,
+    style: props.style,
+    src: props.src,
+    type: props.type,
+    format: props.format,
+    frame: props.frame,
+    onLoad: props.onLoad,
+    onFocus: props.onFocus,
+    onBlur: props.onBlur
+  };
+  if (props.type === 'canvas') {
+    Object.assign(
+      viewerProps,
+      {
+        annotations: JSON.parse(JSON.stringify(props.classification.annotations)),
```

Comment:
Effectively a deep copy. Without this `this.props.annotations` will always equal `nextProps.annotations` inside `shouldComponentUpdate`

Reply:
`annotations: props.classification.annotations.map(annotation => annotation)` maybe?

I'm not entirely sure why you are making a copy of the whole annotations array here.

Reply:
I want to only trigger a re-render when the annotation changes, so think I need to properly deep copy the prop being passed in order to check for changes.
Unfortunately I don't think that'll work as annotations is an array of objects, so the newly constructed array would still refer to the old objects.
i.e.
```javascript
original = [{ a : 1}];
copySlice = original.slice();
copyMap = original.map(i => i);
copyJson = JSON.parse(JSON.stringify(original));
original[0].a = 2;
console.log(original,  copySlice, copyMap, copyJson);
```

Reply:
Sorry, bad phrasing. By re-render I mean update the model, rather than the component.

Reply:
In that case, you should be looking at the annotation for the current task, not all the annotations for the current workflow. That might be passed down as `props.annotation`. If not, it should be the last annotation in the annotations array.